syntax = "proto3";

package p4fuzzer;

import "p4testgen.proto";

service P4FuzzGuide {
    rpc GetP4Coverage(P4CoverageRequest) returns (P4CoverageReply) {}
    rpc RecordP4Testgen(P4CoverageRequest) returns (P4CoverageReply) {}

    rpc GetP4Statement(P4StatementRequest) returns (P4StatementReply) {}
}

message P4CoverageRequest {
    string device_id = 1;
    p4testgen.TestCase test_case = 2;
}

message P4CoverageReply {
    bytes stmt_cov_bitmap = 1;
    int32 stmt_cov_size = 2;
    bytes action_cov_bitmap = 3;
    int32 action_cov_size = 4;
}

message P4StatementRequest {
    int32 idx = 1;
}

message P4StatementReply {
    string statement = 1;
}
